GAE/src/org/akvo/flow/rest/ReportRestService.java

```diff
@@ -0,0 +1,217 @@
+        final Map<String, Object> response = new HashMap<String, Object>();
+        ReportDto dto = null;
+        RestStatusDto statusDto = new RestStatusDto();
+        statusDto.setStatus("failed");
+
+        // if the POST data contains a valid ReportDto, continue.
+        // Otherwise, server will respond with 400 Bad Request
+        if (reportDto != null) {
+            User user = userDao.findUserByEmail(reportDto.getUser());
+
+            if (user != null) {
+                Report r = new Report();
+
+                BeanUtils.copyProperties(reportDto, r, doNotCopy);
+                r.setUser(user.getKey().getId());
+                r.setState(Report.QUEUED);
+                // Save it, so we get an id assigned
+                r = reportDao.save(r);
+
+                //Queue it
+                Queue queue = QueueFactory.getDefaultQueue();
+                TaskOptions options = TaskOptions.Builder.withUrl("/app_worker/reportservlet")
+                        .param(TaskRequest.ACTION_PARAM, ReportTaskRequest.START_ACTION)
+                        .param(ReportTaskRequest.ID_PARAM, Long.toString(r.getKey().getId()));
+                queue.add(options); //overwrite any supplied state
+                dto = new ReportDto();
+                DtoMarshaller.copyToDto(r, dto);
+                statusDto.setStatus("ok");
+            }
+        }
+
+        response.put("meta", statusDto);
+        response.put("report", dto);
+        return response;
+    }
+
+    // find all reports belonging to the current user
+    //TODO: get an unfiltered list if superAdmin?
+    @RequestMapping(method = RequestMethod.GET, value = "")
+    @ResponseBody
+    public Map<String, Object> listMyReports() {
+        final Map<String, Object> response = new HashMap<String, Object>();
+        List<ReportDto> results = new ArrayList<ReportDto>();
+        List<Report> reports = reportDao.listAllByCurrentUser();
+        if (reports != null) {
+            for (Report r : reports) {
+                ReportDto dto = new ReportDto();
+                DtoMarshaller.copyToDto(r, dto);
+                results.add(dto);
+            }
+        }
+        response.put("reports", results);
+        return response;
+    }
+
+    // find a single report by the reportId
+    // TODO: restrict use to owner+superAdmins
+    @RequestMapping(method = RequestMethod.GET, value = "/{id}")
+    @ResponseBody
+    public Map<String, ReportDto> findReport(@PathVariable("id") Long id) {
+        final Map<String, ReportDto> response = new HashMap<String, ReportDto>();
+        Report qo = reportDao.getByKey(id);
+        ReportDto dto = null;
+        if (qo != null) {
+            dto = new ReportDto();
+            DtoMarshaller.copyToDto(qo, dto);
+        }
+        response.put("report", dto);
+        return response;
+
+    }
+
+    // delete report by id
+    @RequestMapping(method = RequestMethod.DELETE, value = "/{id}")
```

Comment:
Deleting the actual file would also need to be handled but it can be in a follow up I created. https://github.com/akvo/akvo-flow/issues/2662